packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -213,19 +216,102 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
     } catch (e) {
     }
 
-    const connectInfo = getConnectInfo(
+    const extraConnectionInfo = getConnectInfo(
       this.uri ? this.uri : '',
       version,
       buildInfo,
       cmdLineOpts,
       topology
     );
 
-    return {
+    this.lastConnectionInfo = {
       buildInfo: buildInfo,
       topology: topology,
-      extraInfo: connectInfo
+      extraInfo: extraConnectionInfo
     };
+    return this.lastConnectionInfo;
+  }
+
+  async getDefaultPrompt(): Promise<string> {
+    return `${await this.getDefaultPromptPrefix()}${this.getTopologySpecificPrompt()}> `;
+  }
+
+  private async getDefaultPromptPrefix(): Promise<string> {
+    if (this.lastConnectionInfo === undefined) {
+      try {
+        await this.getConnectionInfo();
+      } catch (e) {
+        this.lastConnectionInfo = null;
+      }
+    }
+
+    if (this.lastConnectionInfo && this.lastConnectionInfo.extraInfo && this.lastConnectionInfo.extraInfo.is_enterprise) {
+      return 'MongoDB Enterprise ';
+    }
+    return '';
+  }
+
+  private getTopologySpecificPrompt(): string {
+    const description = this.mongoClient.topology?.description;
+    if (!description) {
+      return '';
+    }
+
+    // TODO: replace with proper TopologyType constants - NODE-2973
```

Comment:
Tbh, I think using the strings as-is is just fine – I had understood the problem as us having to do some sort of casting, but since TypeScript would complain about invalid values in the `switch` I think we’re good here

Reply:
Yeah indeed - in the switch it works but it complained in the tests when I tried to use it typed. I'd leave it in there just to be sure to make it consistent.